sys/net/crosslayer/pktbuf/pktbuf.c

```diff
@@ -20,21 +20,24 @@
 #include <string.h>
 
 #include "mutex.h"
+#include "pkt.h"
 #include "pktbuf.h"
 
-#if PKTBUF_SIZE < 128
-typedef uint8_t _pktsize_t;
-#elif PKTBUF_SIZE < 65536
-typedef uint16_t _pktsize_t;
-#else
-typedef size_t _pktsize_t;
+#ifdef DEVELHELP
+#include "od.h"
 #endif
 
-typedef struct __attribute__((packed)) _packet_t {
-    volatile struct _packet_t *next;
+/**
+ * @brief   Data type to represent describe chunks in packet buffer.
+ */
+typedef struct __attribute__((packed)) _pkt_internal_t {
+    struct _pkt_internal_t *next;
     uint8_t processing;
-    _pktsize_t size;
-} _packet_t;
+    pkt_hlist_t *header_ptr;
```

Comment:
I just realized that this might be a problem if you received a packet and want to move the data pointer forward as the headers proceed... -.- I have to rethink the whole `pkt_t` concept now, or at least it's relation with the packet buffer because this also leads to some sever problems in thread-safety...